modules/core/src/main/scala/Json.scala

```diff
@@ -2,129 +2,34 @@ package scalaz
 
 package schema
 
-import scalaz.Scalaz._
-import scalaz.schema.JsonSchema._
-
-object Json {
+trait JsonModule extends SchemaModule {
   type JSON = String
 
-  val module = new SchemaModule {
-    type Prim[A]       = JsonSchema.Prim[A]
-    type ProductTermId = String
-    type SumTermId     = String
-  }
+  type Encoder[A] = A => JSON
 
-  trait ToJson[S[_]] {
-    def serializer: S ~> (? => JSON)
-  }
+  import Schema._
 
-  implicit val toJson: ToJson[module.Prim] = new ToJson[module.Prim] {
-    override def serializer: module.Prim ~> (? => JSON) = new (module.Prim ~> (? => JSON)) {
-      override def apply[A](fa: module.Prim[A]): A => JSON = fa match {
-        case JsonString => a => s""""$a""""
-        case JsonBool   => a => if (a) "true" else "false"
-        case JsonNull   => _ => "null"
-        case JsonNumber => a => a.shows
-      }
-    }
-  }
+  implicit final def algebra(
+    implicit primNT: Prim ~> Encoder,
+    fieldLabel: Liskov[ProductTermId, String],
+    branchLabel: Liskov[SumTermId, String]
```

Comment:
just use `A <~< String` for these. Much more legible.

Reply:
Hold my beer...

Reply:
Done